spark/common/src/test/scala/org/apache/sedona/sql/TestBaseScala.scala

```diff
@@ -95,6 +99,28 @@ trait TestBaseScala extends FunSpec with BeforeAndAfterAll {
     sparkSession.read.format("binaryFile").load(path)
   }
 
+
+  def generateTestData(): Seq[(Int, Double, Geometry)] = {
```

Comment:
Manually generating test data is good. But please do not use randomized data for test because this will create uncertain behaviors in the test and it is really hard to debug. In addition, please make sure it takes a parameter that allows to generate any size of test data as desired.

In addition, when create test data for ST_DWithin join, please make sure you know what the exact number of rows should appear in the result. And you should always use test cases that produce non-zero results.